### PR TITLE
LUKS2 and systemd-boot can be set by the user for the proposal

### PR DESCRIPTION
**Pull request by schubi2**
in the Configuration dialog:

jsc#PED-1906

## Screenshots

Bildschirmfoto_2024-02-16_16-06-43

Other associated PRs :
https://github.com/yast/yast-installation-control/pull/125
https://github.com/yast/yast-bootloader/pull/694
https://github.com/yast/yast-storage-ng/pull/1370

### CODE REVIEW

**Comment by ancorgs** [comment]
I'm not sure whether we want to use the configuration screen to throw all kind of experiments there.

Have you considered using a role with its own client? I know this doesn't fix exactly in the definition of a role, but since the role allows you to modify most of the settings and allows you to write your own client in which you can ask the user whatever you want or present any information I would feel more comfortable with that, as a testing mechanism.

Check for example the Common Criteria dialog.
https://github.com/yast/system-role-common-criteria/blob/master/src/lib/y2common_criteria/dialogs/installation.rb

File: src/lib/installation/console/plugins/luks2_systemd_boot_checkbox.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            # pbkdf2 is default because it can be used by grub2 too
+            proposal_section["encryption"] = { "type" => "luks2", "pbkdf" => "argon2id",
+                                              "password" => Yast::UI.QueryWidget(Id(:pw1), :Value) }
+            ENV["YAST_LUKS2_AVAILABLE"] = "1"
+            Yast::ProductFeatures.SetStringFeature("globals", "prefered_bootloader", "systemd-boot")
+          else
+            proposal_section["encryption"] = {}
+            Yast::ProductFeatures.SetStringFeature("globals", "prefered_bootloader", "")
+          end
+          Yast::ProductFeatures::SetFeature("partitioning", "proposal", proposal_section)
```
Comment (ancorgs) [inline src/lib/installation/console/plugins/luks2_systemd_boot_checkbox.rb:100]: Doesn't this wipe the whole proposal configuration?

File: src/lib/installation/console/plugins/luks2_systemd_boot_checkbox.rb [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+          Yast::UI.ChangeWidget(Id(:pw1), :Value, password)
+        end
+
+        def store
+          proposal_section = Yast::ProductFeatures.GetFeature("partitioning", "proposal")
+
+          if Yast::UI.QueryWidget(Id(:frame), :Value)
+            # pbkdf2 is default because it can be used by grub2 too
+            proposal_section["encryption"] = { "type" => "luks2", "pbkdf" => "argon2id",
+                                              "password" => Yast::UI.QueryWidget(Id(:pw1), :Value) }
```
Comment (ancorgs) [inline src/lib/installation/console/plugins/luks2_systemd_boot_checkbox.rb:93]: I fear storing the password in plain text as part of ProductFeatures may result in password leak in the YaST logs.

Comment (schubi2) [reply src/lib/installation/console/plugins/luks2_systemd_boot_checkbox.rb:93]: Yes, you are right. I have seen how you have fixed this problem:
https://github.com/yast/system-role-common-criteria/commit/feb23e6808aa6aa3217acb7530110f243aeebc04

**Comment by schubi2** [comment]
It is obsolete. Closing

**Pull request closed without merging**